Set `location` and `tags` explicitly to avoid state mismatch

taoufik07 opened the pull request:
When `location` (Same for tags) is not set, the network watcher location is used by default, which causes a mismatch between the deployed infra and the value of the `location` in the state. So every run generates a new plan, that changes the location in terraform's state which's `null`.

![Screenshot 2022-09-13 at 10 53 08](https://user-images.githubusercontent.com/22898667/189869474-8473f90f-5185-4b52-9e3f-92fbc76739dc.png)

taoufik07 committed:
Set `location` and `tags` explicitly to avoid state mismatch

When `location` (Same for tags) is not set, the network watcher
location is used by default, which causes a mismatch between the
deployed infra and the value of the `location` in the state. So every
run generates a new plan, that changes the location in terraform's
state which's `null`.

Shr3ps commented on r-flow-log.tf:
```diff
@@ -1,14 +1,25 @@
   storage_account_id        = var.flow_log_storage_account_id
   enabled                   = var.flow_log_logging_enabled
 
+  location = (
+    var.flow_log_location == null
+    ? data.azurerm_network_watcher.nw.location
+    : var.flow_log_location
+  )
+
```
```suggestion
  location = coalesce(var.flow_log_location, data.azurerm_network_watcher.nw.location)
```

taoufik07 replied:
done

taoufik07 committed:
Use separte storage account for flow_log to avoid lifecycle rules overwrites

Make sure to use a storage account with no existing lifecycle
management rules as this will adds a new rule and overwrites the
existing one.

Fore more details, check https://github.com/hashicorp/terraform-provider-azurerm/issues/6935

taoufik07 committed:
Set the flow_log retention policy days to 91 as recommended

Shr3ps commented on examples/main/modules.tf:
```diff
@@ -83,9 +96,12 @@ module "network_security_group" {
+  flow_log_retention_policy_days    = 91   # default to 91
 
-  flow_log_storage_account_id                    = module.logs.logs_storage_account_id
+  # Make sure to use a storage account with no existing lifecycle management rules
```
Can you add also this note into README?

taoufik07 replied:
done,

taoufik07 replied:
Don't you generate the README? if not I'll add the new var to the README

Shr3ps replied:
I was thinking on a dedicated note in README (not just code comments)

Shr3ps committed:
Update README
